src/main/java/com/google/gcp_variant_transforms/library/SchemaGeneratorImpl.java

```diff
@@ -23,59 +27,133 @@ public TableSchema getSchema(VCFHeader vcfHeader) {
+    }
+    // Adds remaining INFO fields.
+    for (VCFInfoHeaderLine infoHeaderLine : vcfHeader.getInfoHeaderLines()){
+      // INFO header lines with Number = 'A' are added under the alternate bases record
+      if (infoHeaderLine.getCountType() != VCFHeaderLineCount.A)
+        fields.add(convertCompoundHeaderLineToField(infoHeaderLine));
+    }
+    
+    return fields.build();
+  }
+
+  /**
+   * Creates a single Field
+   * @param vcfHeader
+   * @param fieldName
+   * @return field
+   */
+  protected TableFieldSchema createField(VCFHeader vcfHeader, String fieldName){
+    TableFieldSchema field;
+    if (fieldName == Constants.ColumnKeyNames.ALTERNATE_BASES ||
+        fieldName == Constants.ColumnKeyNames.CALLS) {
+      field = createRecordField(vcfHeader, fieldName);
+    } else {
         field = new TableFieldSchema()
             .setName(fieldName)
-            .setMode(SchemaUtils.constantFieldNameToModeMap.get(fieldName))
             .setDescription(SchemaUtils.constantFieldNameToDescriptionMap.get(fieldName))
-            .setFields(callsSubFields)
-            .setType(SchemaUtils.constantFieldNameToTypeMap.get(fieldName));
-      } else {
-        field = new TableFieldSchema()
-            .setName(fieldName)
             .setMode(SchemaUtils.constantFieldNameToModeMap.get(fieldName))
-            .setDescription(SchemaUtils.constantFieldNameToDescriptionMap.get(fieldName))
             .setType(SchemaUtils.constantFieldNameToTypeMap.get(fieldName));
-      }
-      fieldsBuilder.add(field);
     }
-    
-    // Add formats
+    return field;
+  }
 
-    // Add info fields
-    
-    return fieldsBuilder.build();
+  /**
+   * Creates and returns a Field of type Record
+   * @param vcfHeader
+   * @param fieldName
+   * @return a record field
+   */
+  @VisibleForTesting
+  protected TableFieldSchema createRecordField(VCFHeader vcfHeader, String fieldName) {
+    ImmutableList<TableFieldSchema> subFields;
+    if (fieldName == Constants.ColumnKeyNames.ALTERNATE_BASES) {
+      subFields = getAltSubFields(vcfHeader);
+    } else { subFields = getCallSubFields(vcfHeader); } // Call record
+
+    return new TableFieldSchema()
+        .setName(fieldName)
+        .setDescription(SchemaUtils.constantFieldNameToDescriptionMap.get(fieldName))
+        .setMode(SchemaUtils.constantFieldNameToModeMap.get(fieldName))
+        .setFields(subFields)
+        .setType(SchemaUtils.constantFieldNameToTypeMap.get(fieldName));
+  }
+
+  /**
+   * Creates and returns a Field from a compound header line
+   * @param headerLine
+   * @return an INFO or FORMAT Field
+   */
+  @VisibleForTesting
+  protected TableFieldSchema convertCompoundHeaderLineToField(VCFCompoundHeaderLine headerLine) {
+    return new TableFieldSchema()
+        .setName(SchemaUtils.getSanitizedFieldName(headerLine.getID()))
+        .setDescription(headerLine.getDescription())
+        .setMode(SchemaUtils.BQFieldMode.NULLABLE) // Always NULLABLE
```

Comment:
I considered adding this in as another map in SchemaUtils, but this part doesn't lend itself to well that data structure. I have added an if statement, instead, checking if number is <=1 for NULLABLE and setting REPEATED otherwise